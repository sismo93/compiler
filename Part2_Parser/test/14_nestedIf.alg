co test nested if instruction 

begin
  a:=112;
  b:=2;
  if a<20 then
    print(a);
    if a>b*545 then
      print(b)
    else
      if a>5454 then
        print(c)
      endif 
    endif 
  endif
  
  
end
