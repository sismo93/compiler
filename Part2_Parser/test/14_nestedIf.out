Rules used : 1 3 4 7 13 14 15 18 23 21 5 4 7 13 14 15 18 23 21 5 4 8 28 31 34 37 14 15 17 23 21 43 14 15 18 23 21 36 33 3 4 11 47 5 4 8 28 31 34 37 14 15 17 23 21 41 14 15 17 22 26 18 23 21 36 33 3 4 11 47 6 30 3 4 8 28 31 34 37 14 15 17 23 21 41 14 15 18 23 21 36 33 3 4 11 47 6 29 6 6 29 6 
 Here are all the grammar used with more detail : 

<Program> -> begin <Code> end

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Assign>

<Assign> -> [VarName] := <ExprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<InstListPrim> -> SEMICOLON <instList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Assign>

<Assign> -> [VarName] := <ExprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<InstListPrim> -> SEMICOLON <instList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <If>

<If> -> if <cond> then <code> <ifPrim> 

<Cond> -> <andCond> <condPrim>

<andCond> -> <simpleCond> <andCondPrim>

<simpleCond> -> <exprArith> <comp> <exprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> VARNAME

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<comp> -> <

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<andCondPrim> -> EPSILON

<condPrim> -> EPSILON

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Print>

<Print> -> print ( <VarName> )

<InstListPrim> -> SEMICOLON <instList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <If>

<If> -> if <cond> then <code> <ifPrim> 

<Cond> -> <andCond> <condPrim>

<andCond> -> <simpleCond> <andCondPrim>

<simpleCond> -> <exprArith> <comp> <exprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> VARNAME

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<comp> -> >

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> VARNAME

<prodPrim> -> <MultOp> <Atom> <prodPrim>

<multOp> -> *

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<andCondPrim> -> EPSILON

<condPrim> -> EPSILON

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Print>

<Print> -> print ( <VarName> )

<InstListPrim> -> EPSILON

<ifPrim> -> else <Code> endif

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <If>

<If> -> if <cond> then <code> <ifPrim> 

<Cond> -> <andCond> <condPrim>

<andCond> -> <simpleCond> <andCondPrim>

<simpleCond> -> <exprArith> <comp> <exprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> VARNAME

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<comp> -> >

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<andCondPrim> -> EPSILON

<condPrim> -> EPSILON

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Print>

<Print> -> print ( <VarName> )

<InstListPrim> -> EPSILON

<ifPrim> -> endif

<InstListPrim> -> EPSILON

<InstListPrim> -> EPSILON

<ifPrim> -> endif

<InstListPrim> -> EPSILON

